camel/interpreters/e2b_interpreter.py

```
@@ -0,0 +1,119 @@
+            self._sandbox.kill()
+
+    def run(
+        self,
+        code: str,
+        code_type: str,
+    ) -> str:
+        r"""Executes the given code in the e2b sandbox.
+
+        Args:
+            code (str): The code string to execute.
+            code_type (str): The type of code to execute (e.g., 'python',
+                'bash').
+
+        Returns:
+            str: The string representation of the output of the executed code.
+
+        Raises:
+            InterpreterError: If the `code_type` is not supported or if any
+                runtime error occurs during the execution of the code.
+        """
+        if code_type not in self._CODE_TYPE_MAPPING:
+            raise InterpreterError(
+                f"Unsupported code type {code_type}. "
+                f"`{self.__class__.__name__}` only supports "
+                f"{', '.join(list(self._CODE_TYPE_MAPPING.keys()))}."
+            )
+        # Print code for security checking
+        if self.require_confirm:
+            logger.info(
+                f"The following {code_type} code will run on your "
+                "e2b sandbox: {code}"
+            )
+            while True:
+                choice = input("Running code? [Y/n]:").lower()
+                if choice in ["y", "yes", "ye", ""]:
+                    break
+                elif choice not in ["no", "n"]:
+                    continue
+                raise InterpreterError(
+                    "Execution halted: User opted not to run the code. "
+                    "This choice stops the current operation and any "
+                    "further code execution."
+                )
+
+        if self._CODE_TYPE_MAPPING[code_type] is None:
+            execution = self._sandbox.run_code(code)
+        else:
+            execution = self._sandbox.run_code(
+                code=code, language=self._CODE_TYPE_MAPPING[code_type]
+            )
+        return execution.text
```

if execution failed, `execution.text` would be `None`, we should return the error message to user
`return execution.text if execution.text else str(execution.error)`